src/composition/supergraph/config/unresolved/mod.rs

```diff
@@ -0,0 +1,7 @@
+//! Provides objects that represent an unresolved state of a supergraph config.
+
+mod subgraph;
+mod supergraph;
```

Comment:
not to harp too hard on the mod comments, but it'd be nice to know the difference between an unresolved supergraph config and a lazily resolved one; what does the former not have that the latter has?